model_analyzer/config/generate/quick_run_config_generator.py

```diff
@@ -280,26 +282,105 @@ def _get_coordinate_values(self, coordinate: Coordinate,
+                                          perf_analyzer_config)
+
+        if model.model_name() in self._ensemble_submodels:
+            model_run_config.add_ensemble_submodel_configs(ensemble_subconfigs)
+
+        return (model_run_config, model_index)
+
+    def _get_next_ensemble_subconfigs(
+            self, model: ModelProfileSpec,
+            start_model_index: int) -> Tuple[List[ModelConfig], int]:
+        model_index = deepcopy(start_model_index)
+        ensemble_subconfigs = []
+        for ensemble_submodel in self._ensemble_submodels[model.model_name()]:
+            ensemble_subconfigs.append(
+                self._get_next_model_config(ensemble_submodel, model_index))
+            model_index = model_index + 1
 
-        model_variant_name = mc.get_field('name')
-        pac = self._get_next_perf_analyzer_config(model_variant_name, model_num)
+        return (ensemble_subconfigs, model_index)
 
-        model_name = self._models[model_num].model_name()
-        return ModelRunConfig(model_name, mc, pac)
+    def _get_next_ensemble_param_combo(self, start_model_index: int,
+                                       end_model_index: int) -> dict:
+        """
+        For the ensemble model the only parameter we need to set 
+        is the max batch size; which will be the minimum batch size 
+        found in the submodel max batch sizes
+        """
+        min_val_of_max_batch_size = maxsize
+        for model_index in range(start_model_index, end_model_index):
+            dimension_values = self._get_coordinate_values(
+                self._coordinate_to_measure, model_index)
+
+            min_val_of_max_batch_size = int(
+                min([
+                    dimension_values.get("max_batch_size", 1),
+                    min_val_of_max_batch_size
+                ]))
+
+        param_combo = {'max_batch_size': min_val_of_max_batch_size}
+
+        return param_combo
+
+    def _get_next_ensemble_model_config(self, model: ModelProfileSpec,
+                                        ensemble_subconfigs: List[ModelConfig],
+                                        param_combo: dict) -> ModelConfig:
+        model_config = BaseModelConfigGenerator.make_ensemble_model_config(
+            model=model,
+            ensemble_submodel_configs=ensemble_subconfigs,
+            model_variant_name_manager=self._model_variant_name_manager,
+            param_combo=param_combo)
+
+        return model_config
 
-    def _get_next_model_config(self, model_num: int) -> ModelConfig:
+    def _get_next_model_config(self, model: ModelProfileSpec,
+                               dimension_index: int) -> ModelConfig:
         dimension_values = self._get_coordinate_values(
-            self._coordinate_to_measure, model_num)
+            self._coordinate_to_measure, dimension_index)
+
+        if model.cpu_only():
+            kind = "KIND_CPU"
+        elif 'instance_group' in model.get_default_config():
```

Comment:
Is this the behavior we want? I think the old code would put the model on the GPU unless the config told us we can't.

Reply:
Talked w/ Iman and I'm reverting this change